cartographer/mapping_3d/global_trajectory_builder.cc

```diff
@@ -36,9 +36,23 @@ int GlobalTrajectoryBuilder::num_submaps() {
 
 GlobalTrajectoryBuilder::SubmapData GlobalTrajectoryBuilder::GetSubmapData(
     const int submap_index) {
-  return {local_trajectory_builder_->submaps()->Get(submap_index),
-          sparse_pose_graph_->GetSubmapTransform(
-              mapping::SubmapId{trajectory_id_, submap_index})};
+  auto* active_submaps = local_trajectory_builder_->active_submaps();
```

Comment:
For now directly ask the `sparse_pose_graph_` which is guaranteed to know about the submap?